components/nuclio/deploy/component.yaml

```diff
@@ -0,0 +1,28 @@
+name: nuclio deploy
+description: auto build and deploy nuclio function.
+inputs:
+  - {name: Url, type: String, description: 'url/path to source code, zip archive, git path, or notebook'}
+  - {name: Name, type: String, description: 'function name'}
+  - {name: Project, type: String, description: 'project name', default: 'default'}
+  - {name: Tag, type: String, description: 'function version tag', default: ''}
+  - {name: Dashboard, type: String, description: 'nuclio dashboard service url', default: 'http://nuclio-dashboard.nuclio:8070'}
+  - {name: Spec, type: String, description: 'override function spec, Json {key: value, ..}', default: ''}
+  - {name: Env, type: String, description: 'override function env var, Json {key: value, ..}', default: ''}
+  - {name: Mount, type: String, description: 'volume mount, [vol-type:]<vol-url>:<dst-path>', default: ''}
+outputs:
+  - {name: address, type: String, description: 'function endpoint url'}
+implementation:
+  container:
+    image: nuclio/pydeploy
+    command: [
+      python, -m, nuclio, deploy, {inputValue: Url},
+      --dashboard-url, {inputValue: Dashboard},
+      --name, {inputValue: Name},
+      --project, {inputValue: Project},
+      --tag, {inputValue: Tag},
+      --env-json, {inputValue: Env},
+      --spec-json, {inputValue: Spec},
+      --mount, {inputValue: Mount},
+    ]
+    fileOutputs:
+      address: /tmp/output
```

Comment:
JFYI: For future, it's better to have this path passed to your all and not hardcoded.

In
https://github.com/nuclio/nuclio-jupyter/blob/66825220ac659f59d1555ca77568aeecce850e49/nuclio/deploy.py#L69
you should use something like 


```
parser.add_argument('--endpoint-output-file', ...)
...
from pathlib import Path
Path(args.endpoint_output_file).parent.mkdir(parents=True, exist_ok=True)
Path(args.endpoint_output_file).write_text(addr)
```
and then in component.yaml in the `implementation.container.command` array you just add `--endpoint-output-file, {outputPath: address}` and remove the `fileOutputs` section which should only be used as a workaround for programs with hard-coded paths.

Reply:
@Ark-kun the `address` is just an endpoint string (`<addr>:<port>`), not a file, and it is only written once in the life of that container (container will exit when its done), so instead of writing to arbitrary path or loading to external object, i thought its simpler to hardcode and not burden the user.

i can make it an option